python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -1290,7 +1290,7 @@ def _test_write_to_dataset_no_partitions(base_path,
     # Without partitions, append files to root_path
     n = 5
     for i in range(n):
-        pq.write_to_dataset(output_table, base_path,
+        pq.write_to_dataset(output_table, base_path, use_legacy_dataset=True,
```

Comment:
Of course, makes sense.
Yes, I think we should add what you are suggesting to the `pq.write_to_dataset`. Will try and commit for review!

Reply:
Hm, thinking aloud: `existing_data_behavior` controls how the dataset will handle data that already exists. If I implement a unique way of writing parquet files when using the new API in `wrtie_to_dataset` I will also have to set `existing_data_behavior` to be `overwrite_or_ignore`. That will then make trouble when exposing the same parameter in https://issues.apache.org/jira/browse/ARROW-15757.

I could do a check if the parameter is specified or not. But am not sure if there will be additional complications ..

Reply:
So what I don't understand here is that the `dataset.write_dataset` function has a default to raise an error if there is existing data? But then why doesn't the above test fail with that error? (instead of failing in the test because we now overwrote the files)

Reply:
Thinking more about this: if we switch the default as we are now doing, I think we _should_ try to preserve the current behaviour of overwriting/adding data (otherwise it would be a quite breaking change for people using `pq.write_to_dataset` this way). We can still try to deprecate this and later move towards the same default as the dataset.write_dataset implementation. 
But that can be done in a later stage with a proper deprecation warning (eg detect if the directory already exists and is not empty, and in that case indicate this will start raising an error in the future).

Reply:
Yes, good point.
I will step back and redo the issue in a way that the default stays `True` and it raises a deprecation warning in this case.

For the later stage is it worth creating a JIRA already?

Reply:
In this case where the default will still be True, does exposing `write_dataset` keywords still makes sense or should we wait once the default changes?

Reply:
Follow up on this thread: I kept the new implementation as the default and added the use of `basename_template` to mimic the legacy behaviour.

At a later stage I think it will be important to rearrange the keywords for `write_to_dataset` to first list the keywords connected to the new API.

Reply:
With the latest changes, it should be possible to now change the hardcoded `use_legacy_dataset=True` to `use_legacy_dataset=use_legacy_dataset`?